### ISSUE TITLE
IPv6 hostname address not accepted after upgrade to v25.0.0

### ISSUE THREAD

**Issue by jpelletier412** [keycloak/keycloak#32289]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

core

### Describe the bug

After upgrade to v25.0.0, our Keycloak service can no longer start. We have defined KC_HOSTNAME to an IPv6 URL and we get the following error on startup

```
ERROR: Unexpected error when starting the server in (production) mode
ERROR: Failed to start quarkus
ERROR: Provided hostname is neither a plain hostname or a valid URL
```

We have our URL defined as such (ive altered some of the values, but the format is the same):
`https://[1111:2222:3333:4444:5555:0002:0000:0000]/`

This is a valid URL so I am not sure why we are getting this error. I have configured the JVM to have these settings and we still get the error:

```
-Djava.net.preferIPv4Stack=false
-Djava.net.preferIPv6Addresses=true
```
Ive also tried:

```
-Djava.net.preferIPv4Stack=false
-Djava.net.preferIPv6Stack=true
```

This address has worked with previous versions of Keycloak when we defined the URL in KC_HOSTNAME_URL and we have only hit this issue after upgrade to v25.0.0.

### Version

25.0.0

### Regression

- [X] The issue is a regression

### Expected behavior

Keycloak is able to accept the IPv6 URL

### Actual behavior

Keycloak service does not start when an IPv6 URL is provided in the KC_HOSTNAME environment variable

### How to Reproduce?

1. Define IPv6 URL in KC_HOSTNAME
2. Start Keycloak service
3. Notice errors in logs and service does not start

**Comment by jonkoops** [comment]
Did you try the latest patch version of 25? There might be bugs that have since been fixed in this regard.

**Comment by jpelletier412** [comment]
@jonkoops I installed v25.0.4 and the same issue is present

**Comment by jonkoops** [comment]
Thanks for checking, we'll put this back in the triage queue.

**Comment by shawkins** [comment]
Resolved by #32643 as well. Adding a backport for v25 as well.

**Comment by jonkoops** [comment]
@shawkins when doing a backport make sure to add a link to this issue for administrative reasons.
